app/src/main/java/com/google/android/soundchecker/AudioEncoderDecoderActivity.kt

```diff
@@ -851,6 +912,32 @@ class AudioEncoderDecoderActivity : ComponentActivity() {
             for (channel in 0 until numberOfChannels) {
                 mWaveforms!!.add(results[channel].buffer)
             }
+            if (mInputFile == null) {
+                mTopBins = IntArray(results.size)
+                mTopFrequencies = FloatArray(results.size)
+                if (mFundamentalFrequencies == null) {
+                    mFundamentalFrequencies = mutableListOf<MutableList<Float>?>()
+                    for (channel in 0 until results.size) {
+                        mFundamentalFrequencies!!.add(mutableListOf<Float>())
+                    }
```

Comment:
This should be able to be initialized in one line. See https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/-mutable-list.html#

Reply:
Wow, this is so much cleaner!